docs/parts.md

```diff
@@ -0,0 +1,100 @@
+
+# Parts definition
+
+When is necessary to expose external components in the simulator that work together with the board (leds, sensors, speaker, headphone, etc.), it can be done by specifying the JSON definition of the part in the *pxtparts.ts* file using the following arguments:
+
+* **simulationBehavior** *(optional)*: built-in simulator logic name
+* **numberOfPins**: total number of pins including power, GPIO and other pins
```

Comment:
If I were to refactor this definition, I think merging all the pin information into a singe list would be a good goal. Instead, we have three fields: "numberOfPins, "visual.pinLocations", and "pinDefinitions" which must agree in carnality. This makes it easy to get wrong. (Hopefully the allocator has assertions for this, but it may not.) The reason I didn't originally create the schema this way is because of the "visual" object and how that gets threaded through the allocator. It was an expediency for the implementation I had at the time. There is also some benefit to having the "visual" definition be distinct from the behavioral definitions.